js/index.js

```diff
@@ -1,7 +1,86 @@
+
 // Iteration 1: Names and Input
 
+console.log("I'm ready");
+const hacker1 = "Harry Potter"
+console.log("the drivers name is " +  hacker1);
+const hacker2 = "Hermione Lodge"
+console.log("the navigators name is " + hacker2);
 
 // Iteration 2: Conditionals
 
+if (hacker1.length > hacker2.length) 
+{
+  console.log(`The driver has the longest name, it has ${hacker1.length} characters.`)
+}  else if (hacker1.length < hacker2.length) 
+{
+  console.log(`It seems that the navigator has the longest name, it has ${hacker2.length} characters`)
+}
+else 
+{
+  console.log(`Wow, you both have equally long names, ${hacker1.length} characters!`)
+}
+
+const result = []
+for (let i = 0; i < hacker1.length; i++) {
+  result.push(hacker1[i].toUpperCase())
+  result.push(' ')
+}
+console.log(result.join(''))
+
+function reverseString(str) {
+    
+    var splitString = hacker2.split("");  
```

Comment:
You could still use the _str_ parameter in your function and then pass the variable hacker2 as an argument:
`function reverseString(str) {
    var splitString = str.split("");  
    var reverseArray = splitString.reverse(); 
    var joinArray = reverseArray.join(""); 
    return joinArray; 
}
reverseString(hacker2)`

Or remove the parameter _str_ that you are not using now